### PR TITLE
refactor: add `pixi_spec` crate

### PR DESCRIPTION
**Pull request by baszalmstra**
This PR adds the `pixi_spec` crate which introduces `Spec`; an abstraction over `MatchSpec` that can also be used for source dependencies (to be implemented in a follow-up PR).

### CODE REVIEW

File: crates/pixi_spec/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,696 @@
...
+    pub fn try_into_nameless_match_spec(
+        self,
+        root_dir: &Path,
+    ) -> Result<Option<NamelessMatchSpec>, SpecConversionError> {
+        if !self.is_binary() {
+            // Not a binary package
+            return Ok(None);
+        }
+
+        // Convert the path to an absolute path based on the channel config
```
Comment (baszalmstra) [inline crates/pixi_spec/src/lib.rs:532]:
```suggestion
        // Convert the path to an absolute path based on the root_dir
```

File: crates/pixi_manifest/src/feature.rs [hunk 2]
```diff
@@ -7,16 +7,15 @@ use std::{
...
+use pixi_spec::Spec;
+use rattler_conda_types::{PackageName, Platform};
 use rattler_solve::ChannelPriority;
 use serde::{de::Error, Deserialize, Deserializer};
 use serde_with::{serde_as, SerializeDisplay};
 
 use crate::{
     channel::{PrioritizedChannel, TomlPrioritizedChannelStrOrMap},
     consts,
-    parsed_manifest::deserialize_opt_package_map,
```
Comment (baszalmstra) [inline crates/pixi_manifest/src/feature.rs:18]: Bring back

File: crates/pixi_manifest/src/parsed_manifest.rs [hunk 3]
```diff
@@ -403,8 +408,8 @@ mod tests {
             {PROJECT_BOILERPLATE}
             [dependencies]
             test_map = {{ version = ">=1.2.3", channel="conda-forge", build="py34_0" }}
-            test_build = {{ build = "bla" }}
-            test_channel = {{ channel = "conda-forge" }}
+            test_build = {{ version = "*", build = "bla" }}
+            test_channel = {{ version = "*", channel = "conda-forge" }}
```
Comment (baszalmstra) [inline crates/pixi_manifest/src/parsed_manifest.rs:412]: Revert

File: crates/pixi_manifest/src/target.rs [hunk 4]
```diff
@@ -342,16 +338,16 @@ impl<'de> Deserialize<'de> for Target {
         #[serde(deny_unknown_fields)]
         pub struct TomlTarget {
             #[serde(default)]
-            #[serde_as(as = "IndexMap<_, PickFirst<(DisplayFromStr, _)>>")]
-            dependencies: IndexMap<PackageName, NamelessMatchSpec>,
+            // #[serde_as(as = "IndexMap<_, PickFirst<(DisplayFromStr, _)>>")]
```
Comment (baszalmstra) [inline crates/pixi_manifest/src/target.rs:341]:
```suggestion
```

File: crates/pixi_manifest/src/target.rs [hunk 5]
```diff
@@ -342,16 +338,16 @@ impl<'de> Deserialize<'de> for Target {
...
+            dependencies: IndexMap<PackageName, Spec>,
 
             #[serde(default)]
-            #[serde_as(as = "Option<IndexMap<_, PickFirst<(DisplayFromStr, _)>>>")]
-            host_dependencies: Option<IndexMap<PackageName, NamelessMatchSpec>>,
+            // #[serde_as(as = "Option<IndexMap<_, PickFirst<(DisplayFromStr, _)>>>")]
```
Comment (baszalmstra) [inline crates/pixi_manifest/src/target.rs:345]:
```suggestion
```

File: crates/pixi_manifest/src/target.rs [hunk 6]
```diff
@@ -342,16 +338,16 @@ impl<'de> Deserialize<'de> for Target {
...
+            host_dependencies: Option<IndexMap<PackageName, Spec>>,
 
             #[serde(default)]
-            #[serde_as(as = "Option<IndexMap<_, PickFirst<(DisplayFromStr, _)>>>")]
-            build_dependencies: Option<IndexMap<PackageName, NamelessMatchSpec>>,
+            // #[serde_as(as = "Option<IndexMap<_, PickFirst<(DisplayFromStr, _)>>>")]
```
Comment (baszalmstra) [inline crates/pixi_manifest/src/target.rs:349]:
```suggestion
```

File: src/lock_file/satisfiability.rs [hunk 7]
```diff
@@ -351,6 +352,7 @@ pub fn verify_platform_satisfiability(
 }
 
 enum Dependency {
+    Input(rattler_conda_types::PackageName, Spec, Cow<'static, str>),
```
Comment (baszalmstra) [inline src/lock_file/satisfiability.rs:355]: Change to `PixiSpec`.

**Comment by ruben-arts** [comment]
Added errors like this:
image

**Pull request merged**